pset-functions.js

```diff
@@ -91,6 +110,67 @@ console.log('Problem 2:')
 
 // Add your code below this line
 
+/*
+//testing out split array
+console.log(wordCount()[1])
+
+//testing out calling each array
+wordCount().forEach(function(element, index) {
+  console.log(`${element} is at index ${index}`)
+})
+
+//*/
+
+let phrase = "Humpty Dumpty sat on a wall Humpty Dumpty had a great fall"
+
+function wordCount(){
+  return phrase.split(" ")
+}
+
+// Was playing with the stuff above but ran into issues figuring out how to solve, below is form googling
+
+
+function compressArray(original) {
+
+	var compressed = [];
+	// make a copy of the input array
+	var copy = original.slice(0);
+
+	// first loop goes over every element
+	for (var i = 0; i < original.length; i++) {
+
+		var myCount = 0;
+		// loop over every element in the copy and see if it's the same
+		for (var w = 0; w < copy.length; w++) {
+			if (original[i] == copy[w]) {
+				// increase amount of times duplicate is found
+				myCount++;
+				// sets item to undefined
+				delete copy[w];
+			}
+		}
+
+		if (myCount > 0) {
+			var a = new Object();
+			a.value = original[i];
+			a.count = myCount;
+			compressed.push(a);
+		}
+	}
+
+	return compressed;
+};
+
+// It should go something like this:
+
+var newArray = compressArray(wordCount());
+
+console.log(newArray)
+
```

Comment:
@jl1982 Good start to the problem with that function returning the word split is a good helper function for your solution. I see you're getting the outcome of an object, but it's not how the hw output should be. If you get some time, revisit the problem and see if you can take another shot at it.

Take a look at this solution [here](https://github.com/jsr20190826/hw-02-functions-solution/blob/a4d789161886ff115657133c279062df8a4ca72d/pset-functions.js#L103)